src/api/app/models/staging/request_excluder.rb

```diff
@@ -34,11 +26,36 @@ def valid?
 
   private
 
+  def exclude_request(request)
+    bs_request = staging_workflow.target_of_bs_requests.find_by(number: request[:number])
+    return unless valid_request?(bs_request, request[:number])
+
+    request_excluded = Staging::RequestExclusion.new(bs_request: bs_request,
+                                                     number: bs_request.try(:number),
+                                                     description: request[:description],
+                                                     staging_workflow: staging_workflow)
+
+    return if request_excluded.save
+
+    errors << "Request #{request_excluded.bs_request_id}: #{request_excluded.errors.full_messages.to_sentence}."
+  end
+
   def requests_to_be_excluded
     [requests_xml_hash[:request]].flatten
   end
 
   def request_numbers
     [requests_xml_hash[:number]].flatten.map(&:to_i)
   end
+
+  def valid_request?(bs_request, request_number)
+    return true if bs_request && bs_request.staging_project.nil?
+
+    errors << if bs_request
+                "Request #{request_number} could not be excluded because is staged in: #{bs_request.staging_project}"
+              else
+                "Request #{request_number} doesn't exist or it doesn't belong to this project"
```

Comment:
It would be nice to provide a more accurate error message to the user.

To make this possible, we need to retrieve the request and if it exists, check if it belongs to the project. Right now, we only retrieve the request if it belongs to the project, so it's preventing us from knowing if this request simply doesn't exist.

Reply:
True. I will take care of #7355. Can we accept this pull request then, as it is only fixing #8355? Or do you think it is needed to fix #8355 in this same pull request?